add liquidlaunch adapter

lordshisho opened the pull request:
LiquidLaunch is Hyperliquid's first ERC20 fair launch platform on the HyperEVM.

https://x.com/liquidlaunchhl

https://liquidlaunch.app

This is classified as a launchpad

llamabutler commented:
The liquidlaunch adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees liquidlaunch
    
    🦙 Running LIQUIDLAUNCH adapter 🦙
    ---------------------------------------------------
    Start Date:	Sun, 13 Jul 2025 00:06:29 GMT
    End Date:	Mon, 14 Jul 2025 00:06:29 GMT
    ---------------------------------------------------
    
    HYPERLIQUID 👇
    Backfill start time: 1/1/2025
    Daily volume: 5.42 k
    Daily fees: 54
    End timestamp: 1752451588 (2025-07-14T00:06:28.000Z)

lordshisho commented:
> @lordshisho can we track the revenue too? as "All of this revenue to protocol goes to stakers of LIQD" this will be counted as the dailyHoldersRevenue, and dailyRevenue = dailyProtocolRevenue + dailyHolderRevenue and dailyProtocolRevenue will be 0 in this case.

I have added various revenue values to transparently display our fee structure as you requested

I also added RewardAdded events from our staking contract to make it simpler to track what is being provided to who

Supply side revenue is what's paid to token creators that doesn't go to staking... might be better organized into another dimension though

You know best!

llamabutler commented:
The liquidlaunch adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees liquidlaunch
    
    🦙 Running LIQUIDLAUNCH adapter 🦙
    ---------------------------------------------------
    Start Date:	Mon, 14 Jul 2025 14:54:12 GMT
    End Date:	Tue, 15 Jul 2025 14:54:12 GMT
    ---------------------------------------------------
    
    HYPERLIQUID 👇
    Backfill start time: 1/1/2025
    Daily volume: 78.39 k
    Daily fees: 4.06 k
    Daily holders revenue: 2.66 k
    Daily protocol revenue: 0
    Daily supply side revenue: 1.40 k
    Daily revenue: 2.66 k
    End timestamp: 1752591251 (2025-07-15T14:54:11.000Z)
